src/figuresClasses.ts

```diff
@@ -1,11 +1,85 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Triangle sides must be greater than zero');
+    }
+
+    if (a + b <= c || b + c <= a || c + a <= b) {
+      throw new Error(
+        'The longest side must be less than the sum of the other two sides',
+      );
```

Comment:
[CODE STYLE] - The error message is good, but to enhance clarity, you could mention that this is a requirement for a triangle to be valid. For instance: 'For a valid triangle, the length of the longest side must be less than the sum of the other two sides.'